docs/java-helloworld.md

```diff
@@ -0,0 +1,214 @@
+
+<!--SNIPSTART hello-world-project-template-java-activity-interface-->
+<!--SNIPEND-->
+
+Create FormatImpl.java and define the implementation of the Format interface:
+
+<!--SNIPSTART hello-world-project-template-java-activity-->
+<!--SNIPEND-->
+
+### Workflow
+
+Next is our Workflow. Workflow functions are where you configure and organize the execution of Activity functions. Again, the Workflow object is defined like any other, except the interface includes Temporal decorators. Our Workflow has just a single entry method which calls the `composeGreeting()` Activity method and returns the result.
+
+Create HelloWorldWorkflow.java and define the Workflow interface:
+
+<!--SNIPSTART hello-world-project-template-java-workflow-interface-->
+<!--SNIPEND-->
+
+Create HelloWorldWorkflowImpl.java and define the Workflow:
+
+<!--SNIPSTART hello-world-project-template-java-workflow-->
+<!--SNIPEND-->
+
+### Task Queue
+
+Task Queues are how the Temporal server supplies information to Workers. When you start a Workflow, you tell the server which Task Queue the Workflow and/or Activities use as an information queue. We will configure our Worker to listen to the same Task Queue that our Workflow and Activities use. Since the Task Queue name is used by multiple things, let's create Shared.java and define our Task Queue name there:
+
+<!--SNIPSTART hello-world-project-template-java-shared-constants-->
+<!--SNIPEND-->
+
+### Worker
+
+Our Worker hosts Workflow and Activity functions and executes them one at a time. The Worker is instructed to execute the specific functions via information it gets from the Task Queue, and after execution, it communicates results back to the server.
+
+Create HelloWorldWorker.java and define the Worker:
+
+<!--SNIPSTART hello-world-project-template-java-workflow-initiator-->
+<!--SNIPEND-->
+
+### Workflow initiator
+
+There are two ways to start a Workflow, via the Temporal CLI or Temporal SDK. In this tutorial we will use the SDK to start the Workflow which is how most Workflows are started in live environments.
+
+Create InitiateHelloWorld.java and use the SDK to define the start of the Workflow:
+
+<!--SNIPSTART hello-world-project-template-java-workflow-initiator-->
+<!--SNIPEND-->
+
+##  ![](/img/docs/check.png) &nbsp;&nbsp; Test the app
+
+Let's add a simple unit test to our application to make sure things are working as expected. Test code lives in src/test/java/helloworldapp. Gradle will have generated a default AppTest.java in that location. Remove that file and replace it with HelloWorldWorkflowTest.java that contains the following code:
+
+<!--SNIPSTART hello-world-project-template-java-workflow-test-->
+<!--SNIPEND-->
+
+**Terminal**
+
+From the root of the project, run this command:
+
+```
+./gradlew test
+```
+
+**IntelliJ**
+
+From within IntelliJ, right click on HelloWorldWorkflowTest and select Run.
+
+Look for "BUILD SUCCESSFUL" in the output to confirm.
+
```

Comment:
suggestion (take it or leave it): drop the test part to the bottom as a "bonus" section, or clearly mark this part optional so as to reduce the overall length of this tutorial

Reply:
Hmm.. ya good idea. But I think that because this is creating an app from scratch - testing is an essential part of that. For now I will keep it as is and see what kind of feedback we get.